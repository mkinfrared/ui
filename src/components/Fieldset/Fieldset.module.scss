.Fieldset {
  border: 2px solid var(--secondary);
  border-radius: 1rem;
  padding: 2rem;

  &:disabled {
    border-color: var(--grey);

    .legend {
      color: var(--grey);
    }
  }

  .legend {
    color: var(--secondary);
    font-size: 1.8rem;
    padding: 0 1rem;
    text-transform: uppercase;
  }
}
