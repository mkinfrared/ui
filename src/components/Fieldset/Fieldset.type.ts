export type FieldsetProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  children?: React.ReactNode;
  disabled?: boolean;
  legend?: string;
};
