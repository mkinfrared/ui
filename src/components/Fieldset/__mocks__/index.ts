export { default } from "./Fieldset.mock";
