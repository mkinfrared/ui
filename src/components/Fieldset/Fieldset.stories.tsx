import { Meta, Story } from "@storybook/react";
import React from "react";

import Checkbox from "components/Checkbox";
import Radio from "components/Radio";
import TextField from "components/TextField";

import { Fieldset } from "./Fieldset";
import { FieldsetProps } from "./Fieldset.type";
import css from "./Story.module.scss";

export default {
  title: "UI/Fieldset",
  component: Fieldset,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<FieldsetProps> = ({ className, disabled, legend }) => (
  <div className={css.Story}>
    <Fieldset className={className} disabled={disabled} legend={legend}>
      <TextField />
      <Checkbox label="foobar" />
      <Radio label="foobar" />
    </Fieldset>
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
