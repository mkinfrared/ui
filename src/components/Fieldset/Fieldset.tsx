import { classNames } from "utils";

import css from "./Fieldset.module.scss";
import { FieldsetProps } from "./Fieldset.type";

const Fieldset = ({
  className,
  children,
  disabled = false,
  legend,
}: FieldsetProps) => (
  <fieldset
    className={classNames(css.Fieldset, className)}
    data-testid="Fieldset"
    disabled={disabled}
  >
    {legend && <legend className={css.legend}>{legend}</legend>}
    {children}
  </fieldset>
);

export default Fieldset;
