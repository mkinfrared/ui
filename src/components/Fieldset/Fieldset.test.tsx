import { render } from "@testing-library/react";
import React from "react";

import { Fieldset } from "./Fieldset";

describe("<Fieldset />", () => {
  const Component = <Fieldset />;

  it("should be defined", () => {
    expect(Fieldset).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Fieldset");

    expect(element).toBeDefined();
  });

  it("should render a legend tag when 'legend' prop is truthy", () => {
    const legend = "marklar";
    const { container } = render(<Fieldset legend={legend} />);
    const element = container.querySelector("legend");

    expect(element).toBeDefined();
  });
});
