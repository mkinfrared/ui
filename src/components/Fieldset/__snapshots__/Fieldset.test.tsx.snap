// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fieldset /> should match the snapshot 1`] = `
<div>
  <fieldset
    class="Fieldset"
    data-testid="Fieldset"
  />
</div>
`;
